examples/blocked_cholesky_gpu.py

```diff
@@ -175,7 +176,8 @@ async def test_blocked_cholesky():
 
         # Copy and layout input
         a1 = a.copy()
-        ap = a1.reshape(n // block_size, block_size, n // block_size, block_size).swapaxes(1,2)
+        mapper = LDeviceGridBlocked(n // block_size, n // block_size, placement=get_current_devices())
+        ap = mapper.partition_tensor(a1)
```

Comment:
> ... are we using the technique of indexing blocks, then elements?

This was originally used. What we do here is letting the dicing process pass through the partitioner - not arbitrary user reshaping - so that the runtime is aware of the partitioning scheme.

We assume that partitioned blocks are fixed size views of the corresponding part of the original array - users cannot change the partitioning scheme without making another partitioner. To give some background, we used to allow users to arbitrarily partition arrays (with `reshape` etc.) and assume arbitrary accesses including cross-block slicing (e.g. [0:2] for block1 and [2:4] for block2 but accessing [1:3]). We used to try hard to capture this partitioning and slicing behavior at syntax or bytecode level but things became very tricky.